kotlin/display-feature-layers/src/main/java/com/esri/arcgisruntime/sample/displayfeaturelayers/MainActivity.kt

```diff
@@ -0,0 +1,263 @@
+                    FeatureLayerSource.SHAPEFILE -> loadShapefile()
+                }
+                // update the previous feature layer source
+                previousSource = selectedSource
+            }
+        }
+    }
+
+    /**
+     * Load a feature layer with a URL
+     */
+    private fun loadFeatureServiceURL() {
+        // initialize the service feature table using a URL
+        val serviceFeatureTable =
+            ServiceFeatureTable(resources.getString(R.string.sample_service_url)).apply {
+                // set user credentials to authenticate with the service
+                credential = UserCredential("viewer01", "I68VGU^nMurF")
+                // NOTE: Never hardcode login information in a production application
+                // This is done solely for the sake of the sample
+            }
+        // create a feature layer with the feature table
+        val featureLayer = FeatureLayer(serviceFeatureTable)
+        val viewpoint = Viewpoint(41.773519, -88.143104, 4000.0)
+        // set the feature layer on the map
+        setFeatureLayer(featureLayer, viewpoint)
+    }
+
+    /**
+     * Load a feature layer with a portal item
+     */
+    private fun loadPortalItem() {
+        // set the portal
+        val portal = Portal("https://www.arcgis.com", false)
+        // create the portal item with the item ID for the Portland tree service data
+        val portalItem = PortalItem(portal, "1759fd3e8a324358a0c58d9a687a8578")
+        // create the feature layer with the item
+        val featureLayer = FeatureLayer(portalItem)
+        // set the viewpoint to Portland, Oregon
+        val viewpoint = Viewpoint(45.5266, -122.6219, 6000.0)
+        // set the feature layer on the map
+        setFeatureLayer(featureLayer, viewpoint)
+    }
+
+    /**
+     * Load a feature layer with a local geodatabase file
+     */
+    private fun loadGeodatabase() {
+        // locate the .geodatabase file in the device
+        val geodatabaseFile = File(getExternalFilesDir(null), "/LA_Trails.geodatabase")
+        // instantiate the geodatabase with the file path
+        val geodatabase = Geodatabase(geodatabaseFile.path)
+        // load the geodatabase
+        geodatabase.loadAsync()
+        geodatabase.addDoneLoadingListener {
+            if (geodatabase.loadStatus == LoadStatus.LOADED) {
+                // get the feature table with the name
+                val geodatabaseFeatureTable = geodatabase.getGeodatabaseFeatureTable("Trailheads")
+                // create a feature layer with the feature table
+                val featureLayer = FeatureLayer(geodatabaseFeatureTable)
+                // set the viewpoint to Malibu, California
+                val viewpoint = Viewpoint(34.0772, -118.7989, 600000.0)
+                // set the feature layer on the map
+                setFeatureLayer(featureLayer, viewpoint)
+            } else {
+                showError(geodatabase.loadError.message)
```

Comment:
```suggestion
                showError("Error loading geodatabase: " + geodatabase.loadError.message)
```

might offer helpful context